Bring Types and State with elements

BrayonCrayon opened the pull request:
- Retrieves elements relative data for Type/State
- Styled radio buttons ( WIP ) 
- Added functionality to allow user to see the periodic table as a block colour state

BrayonCrayon committed:
Bring Types and State with Elements
- Create interfaces and classes for state/type in elements
- WIP - testing

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
Bring Types and State with Elements
- get type and state from backend
- add tests for type and state types

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>

heyitskei committed:
Fix up failing tests

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>

heyitskei committed:
WIP - Start calculating element's colour code based off of its type

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>

heyitskei committed:
Made colour enum

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>

heyitskei committed:
See colours

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>

heyitskei committed:
We are Canadian!

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>

heyitskei committed:
Bring Types and State with Elements
- get type and state from backend
- add tests for type and state types
- update naming convention to match correct element type for colour
- update state naming convention
- Be able to switch to state display when clicking the button

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Nick Byam <nbyam13@gmail.com>

BrayonCrayon committed:
- Update mocked elements response and add aria-label
- Introduce a button to switch the table colour by type

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Eric Emerson <e.emerson@vehikl.com>
Co-authored-by: Xiao Xue <x.xue@vehikl.com>
Co-authored-by: Felipe Flor <f.flor@vehikl.com>

BrayonCrayon committed:
Update to use radio button group

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Eric Emerson <e.emerson@vehikl.com>
Co-authored-by: Xiao Xue <x.xue@vehikl.com>
Co-authored-by: Felipe Flor <f.flor@vehikl.com>

BrayonCrayon committed:
Get Tests to pass

Co-authored-by: Adam Frank <a.frank@vehikl.com>
Co-authored-by: Segel Mancuso <s.mancuso@vehkl.com>
Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
Refactor to use new custom radio components

BrayonCrayon committed:
WIP - revisit this another day

Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>
Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
- Got radio buttons to look alright and function somewhat.
- Still need to figure out the Worm part (WIP)
- Plan out an attack for colour coding elements by their block

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
- Add block colour to the element class
- Account for hydrogen and helium to use the correct element block colour
- Moved mocked elements api endpoint data into a new folder
- Accounted detached element cases to get the correct block colour
- Added another radio button for the block colour
- Use elements block colour when the display state is of type BLOCK

heyitskei committed:
PR Feedback
- Fix linting issues
- Update test to match correct radio input structure

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>

heyitskei committed:
WIP - Fix tests on colouring by block

Co-authored-by: Nick Whitford <n.whitford@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>

heyitskei committed:
Fix tests on colouring by block

Co-authored-by: CK <c.teoh@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>
Co-authored-by: Brady Deroy <b.deroy@vehikl.com>

BrayonCrayon committed:
- Update table utils test to fix type-linting failures

Co-authored-by: CK <c.teoh@vehikl.com>

BrayonCrayon commented:
@heyitskei so it seems like `typeColour` is working again. I couldn't seem to reproduce the experience you were getting on your laptop last week. ( Screenshot below )

<img width="1891" height="777" alt="image" src="https://github.com/user-attachments/assets/631eb0d7-b26e-4834-a313-df68efb435fb" />

heyitskei commented on src/components/SwitchDisplay.vue:
```diff
@@ -0,0 +1,186 @@
+      <div class="worm__segment"></div>
+      <div class="worm__segment"></div>
+      <div class="worm__segment"></div>
+    </div>
```
Should we consider cleaning the worm logic up in the existing component since we decided to postpone the implementation (and we moved the worm bits into its own component for future reference)?

BrayonCrayon replied:
You make a good point, I'll clean that up and revert it back to a working implementation :)

heyitskei commented on src/components/common/AtomeqRadioInput.vue:
```diff
@@ -0,0 +1,130 @@
+//input
+//  position: fixed
+//  top: -1.5em
+//  left: -1.5em
```
Same here, cleanup the worm? As a housekeeping.

BrayonCrayon replied:
I reverted our implementation for the radio buttons 😃

heyitskei commented on src/types/__tests__/element.spec.ts:
```diff
@@ -0,0 +1,125 @@
+    },
+  );
+
+  // TODO: something ain't quite right here
```
Can we clean this up please?

BrayonCrayon replied:
Comment cleaned up 😃

heyitskei reviewed (changes requested):
Minor changes requested, most non-blocking, but clean up TODOs please.

BrayonCrayon committed:
PR Feedback
- Remove worm radio implementation from actual radio buttons
- Clean up comments from existing code

heyitskei reviewed (approved):
Wonderful! 😄  Thank you for addressing my comments. ❤️  Approved, 🐑 it!
